config/generateRunFile.coffee

```diff
@@ -442,6 +426,116 @@ generateDev = (KONFIG, options) ->
       fi
     }
 
+    function k8s () {
+      params=(start stop)
+      param=$1
+      case "${params[@]}" in  *"$param"*)
+        ;;
+      *)
+        echo "Error: Command not found: $param"
+        echo "Usage: ./run k8s COMMAND"
+        echo ""
+        echo "Commands:  "
+        echo "  start : install the services in Kubernetes pods"
+        echo "  stop  : stop the currently running Kubernetes cluster"
+
+        echo ""
+        exit 1
+      ;;
+      esac
+
+      if [ "$param" == "start" ]; then
+        # Dependency on a hypervisor will be dropped off in the future
+        command -v docker           >/dev/null 2>&1 || { echo >&2 "I require docker but it's not installed.  Aborting."; exit 1; }
+        command -v VirtualBox       >/dev/null 2>&1 || { echo >&2 "I require VirtualBox but it's not installed.  Aborting."; exit 1; }
+        command -v minikube         >/dev/null 2>&1 || { echo >&2 "I require a Kubernetes cluster. To install minikube: \
+                (curl -Lo minikube curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.20.0/minikube-$(uname | awk '{print tolower($0)}')-amd64 && chmod +x minikube && mv minikube /usr/local/bin/)"; exit 1;}
+        command -v kubectl          >/dev/null 2>&1 || { echo >&2 "I require kubectl. To install kubectl: \
+                (curl -L -O https://storage.googleapis.com/kubernetes-release/release/v1.6.4/bin/$(uname | awk '{print tolower($0)}')/amd64/kubectl && chmod +x kubectl && mv kubectl /usr/local/bin/)"; exit 1;}
+        command -v helm             >/dev/null 2>&1 || { echo >&2 "I require helm but it's not installed. To install helm: \
+                (curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get > get_helm.sh && chmod 700 get_helm.sh && ./get_helm.sh)"; exit 1; }
+
+        # minikube must be deleted and recreated if a disk resize is needed
+        minikube start --disk-size 50g
+
+        export MINIKUBE_IP=$(minikube ip)
+        eval $(minikube docker-env)
+
+        docker image inspect baseinstall:v1 >/dev/null 2>&1 && echo yes || echo no >/tmp/response.log 2>&1
+
+        export IMAGE_RESPONSE=$(tail -1 /tmp/response.log)
+
+        if [[ "$IMAGE_RESPONSE" == "no" ]]; then
+          echo 'base installation docker image for koding was not found, building the image:\n'
+          docker build -t baseinstall:v1 -f Dockerfile-k8s-base .
+        fi
+
+        minikube ssh "docker pull mongo:3.2.8"
+        minikube ssh "docker pull koding/postgres:latest"
+        #minikube ssh "docker pull koding/countly-server:latest"
+        minikube ssh "docker pull rabbitmq:3-management"
+        minikube ssh "docker pull redis"
+        minikube ssh "docker pull koding/base"
+
+        kubectl create -f $KONFIG_PROJECTROOT/deployment/kubernetes/backend/backend-deployment.yaml
+
+        export BACKEND_POD_NAME=$(kubectl get pods --namespace default -l "app=backend" -o jsonpath="{.items[0].metadata.name}")
```

Comment:
* better to use a specific namespace instead of the default one
* app=backend is  not a good selector where many other test apps can have "backend" in the default namespace - not an issue once we switch to a dev namespace